lib/ecto/query/builder/distinct.ex

```diff
@@ -54,12 +64,14 @@ defmodule Ecto.Query.Builder.Distinct do
 
   def build(query, binding, expr, env) do
     {query, binding} = Builder.escape_binding(query, binding, env)
-    {expr, {params, _acc}} = escape(expr, {[], %{}}, binding, env)
+    {expr, {params, acc}} = escape(expr, {[], %{subqueries: [], aliases: {:%{}, [], []}}}, binding, env)
```

Comment:
It seems a bit suspicious you are passing are passing the escaped version of `aliases` here. Do we have tests using aliases inside distinct? Should we even allow subqueries/aliases in distinct?